packages/vector_graphics/lib/vector_graphics.dart

```diff
@@ -1,7 +1,96 @@
-library vector_graphics;
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+import 'package:vector_graphics_codec/vector_graphics_codec.dart';
 
-/// A Calculator.
-class Calculator {
-  /// Returns [value] plus 1.
-  int addOne(int value) => value + 1;
+import 'src/listener.dart';
+
+const VectorGraphicsCodec _codec = VectorGraphicsCodec();
+
+/// A widget that displays a vector_graphics formatted asset.
+class VectorGraphicsWidget extends StatefulWidget {
+  const VectorGraphicsWidget.asset({ Key? key, required this.assetName }) : super(key: key);
+
+  /// The name of the asset to be displayed.
+  final String assetName;
+
+  @override
+  State<VectorGraphicsWidget> createState() => _VectorGraphicsWidgetState();
+}
+
+class _VectorGraphicsWidgetState extends State<VectorGraphicsWidget> {
+  ui.Picture? _picture;
+
+  @override
+  void initState() {
+    _loadAssetBytes();
+    super.initState();
+  }
+
+  @override
+  void didUpdateWidget(covariant VectorGraphicsWidget oldWidget) {
+    if (oldWidget.assetName != widget.assetName) {
+      _loadAssetBytes();
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  void dispose() {
+    _picture?.dispose();
+    super.dispose();
+  }
+
+  void _loadAssetBytes() {
+    final AssetBundle bundle = DefaultAssetBundle.of(context);
+    _picture?.dispose();
+    bundle.load(widget.assetName).then((ByteData data) {
+      final FlutterVectorGraphicsListener listener = FlutterVectorGraphicsListener();
+      _codec.decode(data, listener);
+      _picture = listener.toPicture();
```

Comment:
```suggestion
      setState(() {
        _picture = listener.toPicture();
      });
```

Reply:
(But we should jut avoid this whole thing by using a LeafRenderObjectWidget and not having a state at all)